source/matplot/core/axis_type.cpp

```diff
@@ -256,9 +256,9 @@ namespace matplot {
                 if (is_timestamp_) {
                     char buff[20] = {0};
                     time_t now = (time_t)tick_values_[i];
-                    struct tm buf;
-                    localtime_s(&buf, &now);
-                    strftime(buff, 20, tick_label_format_.c_str(), &buf);
+                    struct tm* buf;
```

Comment:
Isn't `#ifdef WIN32 ...` simpler than the external `-D_CRT_SECURE_NO_WARNINGS` here?

Reply:
I've added macros and thread-safe localtime_r to axis_type.cpp, removed the change to CMakeLists.txt, and pushed the changes to my fork. Do I need to do anything else to integrate the changes? Also, when I run the .exe examples on Windows, gnuplot has trouble reading input. Hope you can check this. Thanks for your suggestions.

Reply:
> I've added macros and thread-safe localtime_r to axis_type.cpp, removed the change to CMakeLists.txt, and pushed the changes to my fork. 

Nice. Could you squash these commits?

> Do I need to do anything else to integrate the changes? 

I think we are missing the `std::localtime` default now.

> Also, when I run the .exe examples on Windows, gnuplot has trouble reading input. Hope you can check this. 

Sure. That's unrelated but I'll have a look.

Thanks for the PR.